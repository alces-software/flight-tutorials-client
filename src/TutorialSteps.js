// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/

import React from 'react';

import type { StepMap }  from './types';
import TutorialStep from './TutorialStep';

const TutorialSteps = ({
  completedSteps,
  currentStep,
  steps,
} : {
  completedSteps: Array<string>,
    currentStep: string,
    steps: StepMap,
}) => (
  <div>
    {
      Object.keys(steps).map(stepName => (
        <TutorialStep
          key={stepName}
          completed={completedSteps.includes(stepName)}
          current={stepName === currentStep}
          step={steps[stepName]}
        />
      ))
    }
  </div>
);

export default TutorialSteps;
