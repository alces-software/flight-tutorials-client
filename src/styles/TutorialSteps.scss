/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/

.TutorialStep-title {
    a:hover {
        text-decoration: none;
        color: inherit;
        cursor: inherit;

        .TutorialStep--completed & , .TutorialStep--current & {
            cursor: pointer;
            text-decoration: underline;
        }
    }
    .btn {
        margin-top: -4px;
    }
}
