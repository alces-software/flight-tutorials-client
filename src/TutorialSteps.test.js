/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Compute Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react'
import { render } from 'react-dom'
import { shallow } from 'enzyme';

import TutorialSteps from './TutorialSteps';

const steps = {
  step1: {
    title: 'My step 1',
    description: 'My step 1 description',
  },
  step2: {
    title: 'My step 2',
    description: 'My step 2 description',
  }
};

it('renders without crashing', () => {
  const node = document.createElement('div');
  render(
    <TutorialSteps completedSteps={[]} currentStep="step1" steps={steps} />,
    node
  );
});

Object.keys(steps).forEach((currentStep, idx) => {
  it(`correctly determines the current step (${currentStep})`, () => {
    const wrapper = shallow(
      <TutorialSteps completedSteps={[]} currentStep={currentStep} steps={steps} />
    );

    const renderedSteps = wrapper.find('TutorialStep')

    expect(renderedSteps.at(0)).toHaveProp('current', 0 === idx);
    expect(renderedSteps.at(1)).toHaveProp('current', 1 === idx);
  });
});


it('correctly determines if a step has been completed', () => {
  const wrapper = shallow(
    <TutorialSteps completedSteps={['step1']} currentStep="step2" steps={steps} />
  );

  const renderedSteps = wrapper.find('TutorialStep')

  expect(renderedSteps.at(0)).toHaveProp('completed', true);
  expect(renderedSteps.at(1)).toHaveProp('completed', false);
});

it('renders the steps in order', () => {
  const wrapper = shallow(
    <TutorialSteps completedSteps={[]} currentStep="step1" steps={steps} />,
  );

  const renderedSteps = wrapper.find('TutorialStep')

  expect(renderedSteps.at(0)).toHaveProp('step', steps.step1);
  expect(renderedSteps.at(1)).toHaveProp('step', steps.step2);
});
