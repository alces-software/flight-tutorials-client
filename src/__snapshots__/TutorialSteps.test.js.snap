// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when currentStep === step1 1`] = `
<div
  className="panel-group"
  role="tablist"
>
  <div
    className="TutorialStep TutorialStep--current panel panel-primary"
    id={null}
  >
    <div
      className="panel-heading"
    >
      <div
        className="TutorialStep-title panel-title"
      >
        <a
          aria-controls={undefined}
          aria-expanded={true}
          aria-selected={true}
          className={null}
          href={undefined}
          onClick={[Function]}
          role="tab"
        >
          <span>
            Step 
            1
             
            My step 1
          </span>
          <span
            className="pull-right"
          >
            <button
              className="btn btn-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Skip
            </button>
          </span>
        </a>
      </div>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={false}
      className="panel-collapse collapse in"
      id={undefined}
      role="tabpanel"
    >
      <div
        className="panel-body"
      >
        <div>
          <p>
            My step 1 description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="TutorialStep panel panel-default"
    id={null}
  >
    <div
      className="panel-heading"
    >
      <div
        className="TutorialStep-title panel-title"
      >
        <a
          aria-controls={undefined}
          aria-expanded={false}
          aria-selected={false}
          className="collapsed"
          href={undefined}
          onClick={[Function]}
          role="tab"
        >
          <span>
            Step 
            2
             
            My step 2
          </span>
        </a>
      </div>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={true}
      className="panel-collapse collapse"
      id={undefined}
      role="tabpanel"
    >
      <div
        className="panel-body"
      >
        <div>
          <p>
            My step 2 description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="TutorialStep panel panel-default"
    id={null}
  >
    <div
      className="panel-heading"
    >
      <div
        className="TutorialStep-title panel-title"
      >
        <a
          aria-controls={undefined}
          aria-expanded={false}
          aria-selected={false}
          className="collapsed"
          href={undefined}
          onClick={[Function]}
          role="tab"
        >
          <span>
            Step 
            3
             
            My step 3
          </span>
        </a>
      </div>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={true}
      className="panel-collapse collapse"
      id={undefined}
      role="tabpanel"
    >
      <div
        className="panel-body"
      >
        <div>
          <p>
            My step 3 description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly when currentStep === step2 1`] = `
<div
  className="panel-group"
  role="tablist"
>
  <div
    className="TutorialStep TutorialStep--completed panel panel-success"
    id={null}
  >
    <div
      className="panel-heading"
    >
      <div
        className="TutorialStep-title panel-title"
      >
        <a
          aria-controls={undefined}
          aria-expanded={false}
          aria-selected={false}
          className="collapsed"
          href={undefined}
          onClick={[Function]}
          role="tab"
        >
          <span>
            Step 
            1
             
            My step 1
          </span>
        </a>
      </div>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={true}
      className="panel-collapse collapse"
      id={undefined}
      role="tabpanel"
    >
      <div
        className="panel-body"
      >
        <div>
          <p>
            My step 1 description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="TutorialStep TutorialStep--current panel panel-primary"
    id={null}
  >
    <div
      className="panel-heading"
    >
      <div
        className="TutorialStep-title panel-title"
      >
        <a
          aria-controls={undefined}
          aria-expanded={true}
          aria-selected={true}
          className={null}
          href={undefined}
          onClick={[Function]}
          role="tab"
        >
          <span>
            Step 
            2
             
            My step 2
          </span>
          <span
            className="pull-right"
          >
            <button
              className="btn btn-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Skip
            </button>
          </span>
        </a>
      </div>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={false}
      className="panel-collapse collapse in"
      id={undefined}
      role="tabpanel"
    >
      <div
        className="panel-body"
      >
        <div>
          <p>
            My step 2 description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="TutorialStep panel panel-default"
    id={null}
  >
    <div
      className="panel-heading"
    >
      <div
        className="TutorialStep-title panel-title"
      >
        <a
          aria-controls={undefined}
          aria-expanded={false}
          aria-selected={false}
          className="collapsed"
          href={undefined}
          onClick={[Function]}
          role="tab"
        >
          <span>
            Step 
            3
             
            My step 3
          </span>
        </a>
      </div>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={true}
      className="panel-collapse collapse"
      id={undefined}
      role="tabpanel"
    >
      <div
        className="panel-body"
      >
        <div>
          <p>
            My step 3 description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly when currentStep === step3 1`] = `
<div
  className="panel-group"
  role="tablist"
>
  <div
    className="TutorialStep TutorialStep--completed panel panel-success"
    id={null}
  >
    <div
      className="panel-heading"
    >
      <div
        className="TutorialStep-title panel-title"
      >
        <a
          aria-controls={undefined}
          aria-expanded={false}
          aria-selected={false}
          className="collapsed"
          href={undefined}
          onClick={[Function]}
          role="tab"
        >
          <span>
            Step 
            1
             
            My step 1
          </span>
        </a>
      </div>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={true}
      className="panel-collapse collapse"
      id={undefined}
      role="tabpanel"
    >
      <div
        className="panel-body"
      >
        <div>
          <p>
            My step 1 description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="TutorialStep TutorialStep--completed panel panel-success"
    id={null}
  >
    <div
      className="panel-heading"
    >
      <div
        className="TutorialStep-title panel-title"
      >
        <a
          aria-controls={undefined}
          aria-expanded={false}
          aria-selected={false}
          className="collapsed"
          href={undefined}
          onClick={[Function]}
          role="tab"
        >
          <span>
            Step 
            2
             
            My step 2
          </span>
        </a>
      </div>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={true}
      className="panel-collapse collapse"
      id={undefined}
      role="tabpanel"
    >
      <div
        className="panel-body"
      >
        <div>
          <p>
            My step 2 description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="TutorialStep TutorialStep--current panel panel-primary"
    id={null}
  >
    <div
      className="panel-heading"
    >
      <div
        className="TutorialStep-title panel-title"
      >
        <a
          aria-controls={undefined}
          aria-expanded={true}
          aria-selected={true}
          className={null}
          href={undefined}
          onClick={[Function]}
          role="tab"
        >
          <span>
            Step 
            3
             
            My step 3
          </span>
        </a>
      </div>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={false}
      className="panel-collapse collapse in"
      id={undefined}
      role="tabpanel"
    >
      <div
        className="panel-body"
      >
        <div>
          <p>
            My step 3 description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
